aptos-move/block-executor/src/view.rs

```diff
@@ -2541,7 +2542,7 @@ mod test {
 
     fn create_state_value(value: &Value, layout: &MoveTypeLayout) -> StateValue {
         StateValue::new_legacy(
-            ValueSerDeContext::new()
+            ValueSerDeContext::new(None)
```

Comment:
nit: Might be good to add in each case where `max_value_nested_depth` is `None`, why it is okay (mock, tests). Could also be done as a followup.

Reply:
Yes, these tests are very messy and depth is not very relevant here, let's iterate over those separately